commands/api-gateway/gateway/customDomains/remove.go

```diff
@@ -0,0 +1,63 @@
+package customDomains
+
+import (
+	"context"
+	"fmt"
+	"github.com/ionos-cloud/ionosctl/v6/commands/api-gateway/completer"
+	"github.com/ionos-cloud/ionosctl/v6/internal/client"
+	"github.com/ionos-cloud/ionosctl/v6/internal/constants"
+	"github.com/ionos-cloud/ionosctl/v6/internal/core"
+	"github.com/ionos-cloud/sdk-go-bundle/products/apigateway/v2"
+	"github.com/spf13/viper"
+)
+
+func RemovetCmd() *core.Command {
+	cmd := core.NewCommand(context.Background(), nil, core.CommandBuilder{
+		Namespace: "apigateway",
+		Resource:  "customdomains",
+		Verb:      "remove",
+		Example:   "ionosctl apigateway gateway customdomains remove --gateway-id ID --custom-domains-id ID",
+		PreCmdRun: func(c *core.PreCommandConfig) error {
+			if err := core.CheckRequiredFlags(c.Command, c.NS, constants.FlagGatewayID, constants.FlagCustomDomainsId); err != nil {
+				return err
+			}
+
+			return nil
+		},
+		CmdRun: func(c *core.CommandConfig) error {
+			apigatewayId := viper.GetString(core.GetFlagName(c.NS, constants.FlagGatewayID))
+			customDomainsId := viper.GetInt(core.GetFlagName(c.NS, constants.FlagCustomDomainsId))
+			usedApiGateway, _, _ := client.Must().Apigateway.APIGatewaysApi.ApigatewaysFindById(context.Background(), apigatewayId).Execute()
+			input := usedApiGateway.Properties
+			if input.CustomDomains == nil {
+				fmt.Errorf("There are no custom domains defined in this apigateway!")
+			} else if customDomainsId == 0 {
+				input.CustomDomains = input.CustomDomains[1:]
+			} else if customDomainsId == 5 {
+				input.CustomDomains = input.CustomDomains[:len(input.CustomDomains)-1]
+			} else {
+				input.CustomDomains = append(input.CustomDomains[:customDomainsId], input.CustomDomains[customDomainsId+1:]...)
+			}
```

Comment:
1. this logic is generally brittle and hardcoded and the error is not properly returned. lets rewrite the error logic:

```
if customDomainsId < 0 || customDomainsId >= len(input.CustomDomains) {
    return fmt.Errorf("Invalid custom domain index")
}
```

2. for removing a custom domain, you can keep the last bit intact. Just remove intermediary magic numbers as those are not needed.